server/plugins/RawGeoFromRegex.js

```diff
@@ -0,0 +1,121 @@
+/*
+Regex Replace
+Allows matching and replacing
+*/
+var db = require('../knex/knex.js');
+var OSPoint = require('ospoint');
+var getJSON = require('get-json')
+var logger = require('../log');
+var http = require('http');
+
+function run(trigger, scope, data, config, callback) {
+    // logger.main.error('Configgerino: ' + JSON.stringify(config));
+    if (data.agency) {
+    	logger.main.error('It has an agency! Which is - ' + data.agency)
+    	/* Grab filter from db for agency defined for this message, 
+    	* then if one exists, whack raw_geolocation into the return data.
+    	* Simples.
+   		*/
+        // data.raw_geolocation = 0;
+        var filterConfig = false;
+        for (var i = 0; i <= (config.filters.length-1); i++) {
+            var currentFilter = config.filters[i];
+            if (currentFilter.agency == data.agency) {
+                filterConfig = currentFilter;
+            }
+        }
+
+        if (filterConfig) {
+            logger.main.error('before the normalize check');
+            
+            data.raw_geolocation = data.message.match(new RegExp(filterConfig.regex))[0];
+
+            logger.main.error('RAW GEOLOCATION -- ' + JSON.stringify(data.raw_geolocation));
+
+            //TODO - rename raw to processed lol
+            data.coords = normalizeAddressData(
+                    data.raw_geolocation, 
+                    filterConfig.flags,
+                    config.location_services.api_key,
+                    data.id
+            );
+            // logger.main.error('after the normalize check - ' + JSON.stringify(data));
+        }
+	}
+
+    // logger.main.error("GEOLOCATION AFTER LOOP - " + data.raw_geolocation);
+
+	logger.main.error('data object before giving to CALLBACK : ' + JSON.stringify(data));
+
+
+    callback(data);
+}
+
+
+function updateDB(raw, coords, messageID) {
+    logger.main.error('RAW: ' + raw + '; Coords: ' + coords + '; messageID: ' + messageID);
+
+    db.from('messages')
+        .where('id', '=', messageID)
+        .update({
+            raw_geolocation : raw,
+            coords : coords
+        })
+        .then ((result) => { 
+            logger.main.error('RESULT OF DB OPERATION: ' + JSON.stringify(result));
+        })
+        .catch ((err) => {
+            logger.main.error('GEO DB ERROR = ' + err);
+        });
+    // logger.main.error('filtered message = ' + data.raw_geolocation);
+    return coords;
+}
+
+function getCoordsFromAddress(address, apiKey, messageID) {
+    getJSON('https://eu1.locationiq.com/v1/search.php?key=' + apiKey + '&q=' + address +'&format=json', 
```

Comment:
The getJSON lib seems unnecessary here - there was also a commit ahead of your branch that moved all plugins to use `axios`, so probably best to switch to using that